ee/app/livechat-enterprise/server/hooks/beforeRoutingChat.js

```diff
@@ -3,10 +3,29 @@ import { settings } from '../../../../../app/settings/server';
 import { LivechatInquiry } from '../../../../../app/models/server';
 import { dispatchInquiryPosition } from '../lib/Helper';
 import { allowAgentSkipQueue } from '../../../../../app/livechat/server/lib/Helper';
+import { Livechat } from '../../../../../app/livechat/server/lib/Livechat';
+import { LivechatDepartment, LivechatInquiry as LivechatInquiryRaw, LivechatRooms } from '../../../../../app/models/server/raw';
+import { online } from '../../../../../app/livechat/server/api/lib/livechat';
 import { saveQueueInquiry } from '../../../../../app/livechat/server/lib/QueueManager';
 import { cbLogger } from '../lib/logger';
 
 callbacks.add('livechat.beforeRouteChat', async (inquiry, agent) => {
+	// check here if department has fallback before queueing
+	if (inquiry?.department && !online(inquiry.department, true)) {
+		cbLogger.debug('No agents online on selected department. Inquiry will use fallback department');
```

Comment:
Wondering if we should add some system message here? something like "the tranfer to original department X failed, due to which it was transferred to fallback department Y"